New functionality for N-1 histograms, Including dimuon categories for cutflows, Changed all Selections to Cuts

kerstinlovisa opened the pull request:
I've changed the n-1 histograms so I only fill the histograms corresponding to the cut not applied. It uses the same names of the cuts specified in the config, same as for the TTAlpsDimuonSelections.

Some other changes:
Including the dimuon categories (pat-pat, pat-dsa, dsa-dsa) in the cutflows 
I changed all Selections to Cuts as requested in the last PR
including new signal cross sections
small changes in objects selections

kerstinlovisa commented:
Changed Selections to Cuts in some tea files as well, see PR in tea: https://github.com/jniedzie/tea/pull/20

jniedzie commented on libs/user_extensions/src/TTAlpsHistogramFiller.cpp:
```diff
@@ -671,8 +670,33 @@ void TTAlpsHistogramFiller::FillLLPnanoAODLooseMuonsVertexHistograms(const share
-    FillMuonVertexHistograms(event, collectionName);
+    // Only for the BestVertex collections
+    if(collectionName.find("Best") == string::npos) continue;
+    for(auto cut : muonVertexCollections[collectionName]) {
+
+      if(cut.find("Cut") == string::npos) continue;
```
I'm confused by what "cut" is here, given it comes from the muon vertex collection. I guess there may be some naming issue here. Also, what does the "if" statement do? Maybe this line deserves a short comment

kerstinlovisa replied:
I changed it now, it was to avoid running on the cut named "BestDimuonVertex" as it's not a variable

kerstinlovisa commented on configs/ttalps_plotter_signal_config.py:
```diff
@@ -184,46 +185,34 @@
-      # Histogram(muonVertexCollectionName+category+"_pfRelIso04all2"                  , "", False, True  , default_norm        , 4  , 0     , 1    , 1e-3  , 1e6   , "#mu_{2} I_{PF}^{rel} ( #Delta R < 0.4 )"    , "# events (2018)"   ),
-      # Histogram(muonVertexCollectionName+category+"_tkRelIsoMuon1"                   , "", False, True  , default_norm        , 4  , 0     , 10    , 1e-3  , 1e6   , "#mu_{1} I_{tk}^{rel} ( #Delta R < 0.3 )"    , "# events (2018)"   ),
-      # Histogram(muonVertexCollectionName+category+"_tkRelIsoMuon2"                   , "", False, True  , default_norm        , 4  , 0     , 10    , 1e-3  , 1e6   , "#mu_{2} I_{tk}^{rel} ( #Delta R < 0.3 )"    , "# events (2018)"   ),
+      # Histogram(muonVertexCollectionName+category+"_dxyPVTraj1"           , "", False, True  , default_norm        , 10 , 0     , 800   , 1e-3  , 1e6   , "#mu vertex d_{xy}^{1} [cm]"             , "# events (2018)"   ),
+      # Histogram(muonVertexCollectionName+category+"_dxyPVTraj2"           , "", False, True  , default_norm        , 20 , 0     , 800   , 1e-3  , 1e6   , "#mu vertex d_{xy}^{2} [cm]"             , "# events (2018)"   ),
+      # Histogram(muonVertexCollectionName+category+"_dxyPVTrajSig1"        , "", False, True  , default_norm        , 2  , 0     , 80    , 1e-3  , 1e6   , "#mu vertex d_{xy}^{1} / #sigma_{dxy}^{1}"  , "# events (2018)"   ),
+      # Histogram(muonVertexCollectionName+category+"_dxyPVTrajSig2"        , "", False, True  , default_norm        , 2  , 0     , 80    , 1e-3  , 1e6   , "#mu vertex d_{xy}^{2} / #sigma_{dxy}^{2}"  , "# events (2018)"   ),
```
I just want to keep this commented out so I don't have to redo the binning in the future

kerstinlovisa commented:
I have included the code for the new variables I introduced before christmas, when I compared to variables in the dispalced dimuon search, such as the alpha variables. 

I also added another flag for the histogramming: runGenMuonVertexCollectionHistograms which should have the muonVertexCollection histograms matched to gen-level

kerstinlovisa commented:
For some of these new variables I also defined functions in NanoDimuonVertex in tea in this PR:
https://github.com/jniedzie/tea/pull/21

kerstinlovisa commented on libs/user_extensions/src/TTAlpsCuts.cpp:
```diff
@@ -21,31 +21,36 @@ TTAlpsSelections::TTAlpsSelections(){
+    if(dimuonCategory != "") collectionName = originalCollectionName + "_" + dimuonCategory;
     cutFlowManager->RegisterCollection(collectionName);
     cutFlowManager->RegisterCut("initial", collectionName);
+    cutFlowManager->RegisterCut("InitalDimuon", collectionName);
```
typo fixed and lower case

kerstinlovisa commented on configs/ttalps_object_cuts.py:
```diff
@@ -121,7 +64,8 @@
 JPsiDimuonVertexBaseCuts = {
   "maxDCA": 15.0,
-  "maxChargeProduct": 0.0,
+  # "maxChargeProduct": 0.0,
```
change it to -1

jniedzie replied:
Wait, but the charge product for the opposite sign is supposed to be -1, so isn't that dangerous, too? That's why I proposed -0.1 😅

kerstinlovisa replied:
Ah I misread, right I can change it

kerstinlovisa commented on configs/ttalps_histogrammer_signal_config.py:
```diff
@@ -223,6 +223,55 @@
+    # (muonVertexCollectionName , "OuterDeltaR"              , 500    , 0      , 10    , ""  ),
+    # (muonVertexCollectionName , "ProxDeltaR"               , 500    , 0      , 10    , ""  ),
+  )
+  for category in muonVertexCollectionCategories:
```
I merged some code, I had to add allDimuonVertexCategories which has the dimuon categories+"" as options
